src/sentry/release_health/metrics.py

```diff
@@ -677,396 +683,399 @@ def check_releases_have_health_data(
+
+        return ret_val
 
     @staticmethod
     def _get_errored_sessions_for_overview(
+        projects: Sequence[Project],
         where: List[Condition],
         org_id: int,
-        rollup: int,
+        granularity: int,
+        start: datetime,
+        end: datetime,
     ) -> Mapping[Tuple[int, str], int]:
         """
         Count of errored sessions, incl fatal (abnormal, crashed) sessions,
         excl errored *preaggregated* sessions
         """
-        rv_errored_sessions: Dict[Tuple[int, str], int] = {}
+        project_ids = [p.id for p in projects]
 
-        release_column_name = resolve_tag_key(USE_CASE_ID, org_id, "release")
-        aggregates: List[SelectableExpression] = [
-            Column(release_column_name),
-            Column("project_id"),
+        select = [
+            MetricField(metric_mri=SessionMRI.ERRORED_SET.value, alias="value", op=None),
         ]
 
-        for row in raw_snql_query(
-            Request(
-                dataset=Dataset.Metrics.value,
-                app_id=SnubaAppID,
-                query=Query(
-                    match=Entity(EntityKey.MetricsSets.value),
-                    select=aggregates + [Function("uniq", [Column("value")], "value")],
-                    where=where
-                    + [
-                        Condition(
-                            Column("metric_id"),
-                            Op.EQ,
-                            resolve(USE_CASE_ID, org_id, SessionMRI.ERROR.value),
-                        ),
-                    ],
-                    groupby=aggregates,
-                    granularity=Granularity(rollup),
-                ),
-            ),
-            referrer="release_health.metrics.get_errored_sessions_for_overview",
-        )["data"]:
-            key = row["project_id"], reverse_resolve(USE_CASE_ID, org_id, row[release_column_name])
-            rv_errored_sessions[key] = row["value"]
+        groupby = [
+            MetricGroupByField(field="project_id"),
+            MetricGroupByField(field="release"),
+        ]
+
+        query = MetricsQuery(
+            org_id=org_id,
+            project_ids=project_ids,
+            select=select,
+            start=start,
+            end=end,
+            granularity=Granularity(granularity),
+            groupby=groupby,
+            where=where,
+            include_series=False,
+            include_totals=True,
+        )
+        raw_result = get_series(
+            projects=projects,
+            metrics_query=query,
+            use_case_id=USE_CASE_ID,
+        )
+        groups = raw_result["groups"]
+
+        ret_val = {}
 
-        return rv_errored_sessions
+        for group in groups:
+            by = group.get("by", {})
+            proj_id = by.get("project_id")
+            release = by.get("release")
+
+            value = get_path(group, "totals", "value")
+            ret_val[(proj_id, release)] = value
+        return ret_val
```

Comment:
not related: All these hardcoded transformations of the results are very error prone, I would suggest in the future to maybe try to find a solution to make it as transparent as possible.

Reply:
I agree that they are error prone, I had to look at what we were generating before to do it but I'm not sure I get your suggestion.
Transparent for who ?
I could of course encapsulate the processing in a function but I don't think it will make it less error prone, and I saw little point of doing it since it was not reused anywhere.

Reply:
agree with @RaduW. All these small pieces are unique, so it's hard to find common lines.